packages/react-next/src/components/Checkbox/useCheckboxClasses.tsx

```diff
@@ -0,0 +1,81 @@
+import { ICheckboxProps, ICheckboxStyleProps, ICheckboxStyles } from './Checkbox.types';
+import { memoizeFunction, css, useClasses } from '../../Utilities';
+import { ITheme, getGlobalClassNames } from '../../Styling';
+import * as classes from './Checkbox.scss';
+
+export function useCheckboxClasses(props: ICheckboxProps): { [key in keyof ICheckboxStyles]: string } {
+  return useClasses<ICheckboxProps['styles'], ICheckboxStyleProps, ICheckboxStyles>({
+    customizationScopeName: 'Checkbox',
+    useStaticStyles: true,
+    stylesProp: props.styles,
+    styleProps: {
+      className: props.className,
+      disabled: props.disabled,
+      indeterminate: props.indeterminate,
+      checked: props.checked,
```

Comment:
if the checkbox is uncontrolled, how is the uncontrolled `checked` state accessed? (In other words, the input to `useCheckboxClasses` should be `state: ICheckboxState` which in many (most) cases is just the same as the props with perhaps a few omissions and adds.

Reply:
But overall I think this is the right approach. I actually think this would enable us to replace `styled` with `compose` to glue them together. All the styling logic goes into the hook.

```tsx
const DetailsList = compose(DetailslListBase, {
  classes: useDetailsListClasses
});
```

Reply:
One other note, in trying to glue v0 and v7 together, in v0, the resolution of classes returns something different.

Here, your hook returns

```
{
  root: 'ms-Toggle root-123',
  pill: 'pill-124',
  etc
}
```

In v0, they need this:
```tsx
{
  classes: { ..same as above },
  style: {
     root: { background: 'red', ... }
     pill: { ... }
  }
}
```